File: src/SSHServer/Compressions/NoCompression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SSHServer.Compressions
{
    public class NoCompression : ICompression
    {
        public string Name
        {
            get
            {
                return "none";
            }
        }

        public byte[] Compress(byte[] data)
        {
            return data;
        }

        public byte[] Decompress(byte[] data)
        {
            return data;
        }
    }
}

File: SSHServerNodeJS/SSHServerNodeJS/Compressions/NoCompression.ts
import { ICompression } from "./ICompression";

export class NoCompression implements ICompression {
    public getName(): string {
        return "none";
    }

    public compress(data: Buffer): Buffer {
        return data;
    }

    public decompress(data: Buffer): Buffer {
        return data;
    }
}
